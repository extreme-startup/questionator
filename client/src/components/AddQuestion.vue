<template>
  <div>
    <slot :clickHandler="show"></slot>
    <Modal v-show="isShown" @close="close">
      <template v-slot:header>
        <Title>Add new question</Title>
      </template>
      <QuestionForm
        v-model="question"
        :typeOptions="typeOptions"
        :errors="errors"
        placeholder="Select question type"
        @submit="submitQuestion"
        @close="close"
      ></QuestionForm>
    </Modal>
  </div>
</template>

<script>
import styled from 'vue-styled-components';
import Modal from '@/common/Modal.vue';
import QuestionForm from '@/components/QuestionForm.vue';
import { TitleH3 } from '@/common/styledComponents';
import { toggle } from '@/common/mixins/toggle';

// TODO: remove mock data
const mockedOptions = [{ id: 1, value: 'Static' }, { id: 2, value: 'Dynamic' }];

const Title = styled(TitleH3)`
  width: 100%;
  padding: 30px 0 30px 60px;
  margin: 0;
  text-align: left;
  font-weight: normal;
  box-sizing: border-box;
  background-color: var(--bg-color);
`;

const defaults = {
  type: '',
  text: '',
  answer: '',
  value: '',
};

const errors = {
  type: {
    required: false,
  },
  text: {
    required: false,
  },
  answer: {
    required: false,
  },
};

export default {
  name: 'addQuestion',
  data: () => ({
    typeOptions: mockedOptions,
    question: { ...defaults },
    errors,
  }),
  components: {
    Modal,
    Title,
    QuestionForm,
  },
  mixins: [toggle],
  methods: {
    async submitQuestion(question) {
      this.question = { ...defaults };
      this.close();
      try {
        // TODO: check post req
        await this.$http.post('/question', question);
      } catch (err) {
        // TODO: log error
      }
    },
  },
};
</script>
