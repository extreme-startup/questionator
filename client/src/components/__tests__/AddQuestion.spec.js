import { shallowMount } from '@vue/test-utils';
import AddQuestion from '../AddQuestion.vue';

const typeOptions = [
  {
    id: 0,
    value: 'option 1',
  },
  {
    id: 1,
    value: 'option 2',
  },
];

const errors = {
  type: {
    required: false,
  },
  text: {
    required: false,
  },
  answer: {
    required: false,
  },
};

const question = {
  type: '',
  text: '',
  answer: '',
  value: '',
};

describe('AddQuestion', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = shallowMount(AddQuestion, {
      propsData: {
        typeOptions,
        errors,
        question,
      },
    });
  });

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
