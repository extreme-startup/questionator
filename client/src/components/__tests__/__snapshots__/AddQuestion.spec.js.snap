// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddQuestion renders correctly 1`] = `
<div
  errors="[object Object]"
  question="[object Object]"
  typeoptions="[object Object],[object Object]"
>
   
  <modal-stub
    style="display: none;"
  >
     
    <questionform-stub
      errors="[object Object]"
      placeholder="Select question type"
      typeoptions="[object Object],[object Object]"
      value="[object Object]"
    />
  </modal-stub>
</div>
`;
